data.tf

```diff
@@ -1,3 +1,10 @@
+data "external" "get_eclwatchip" {
+  program = ["./scripts/get_eclwatchip.pl"]
```

Comment:
ECL Watch's IP address can be obtained without executing an external Perl script.  Terraform itself can execute things and collect the output.  See resource.null_resource.az_graph in main.tf as an example invocation.

Reply:
> ECL Watch's IP address can be obtained without executing an external Perl script. Terraform itself can execute things and collect the output. See resource.null_resource.az_graph in main.tf as an example invocation.

I found out the "resource.null" doesn't support output, which means it won't support returning the ecl wath ip.